### PR TITLE
updated spring context for keycloak

### PR DESCRIPTION
**Pull request by taba90**
## Description

- Contains spring-security-context changes to enable keycloak
- Some improvements to login UI for multiple login entries (google, keycloak)
- Contains documentation improvemnts and migration guidelines

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#<issue>

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by offtherailz — CHANGES_REQUESTED** [review]
- [ ] wait #7872 
- [ ] document integration (openid.md)
- [ ] Update migration guidelines (actually changes are only the ones to support google)

**Review by offtherailz — APPROVED** [review]

File: docs/developer-guide/integrations/users/openId.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# Integration with OpenID connect
+
+MapStore allows to integrate and login using some common [OpenID connect](https://openid.net/connect/) services. With these configurations, you can add to the login form some buttons to login with the given service.
+
+
+## Customizing logo an text in Login Form
+
+For details about the configuration for a specific service, please refer to the specific section below. For details about `authenticationProviders` optional values (e.g. to customize icon and/or text to show), refer to the documentation of the LoginPlugin.
```
Comment (tdipisa) [inline docs/developer-guide/integrations/users/openId.md:8]: We should put a link to the LoginPlugin here.

File: docs/developer-guide/integrations/users/openId.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
+# Integration with OpenID connect
+
+MapStore allows to integrate and login using some common [OpenID connect](https://openid.net/connect/) services. With these configurations, you can add to the login form some buttons to login with the given service.
```
Comment (tdipisa) [inline docs/developer-guide/integrations/users/openId.md:3]: I would say as follow:

"MapStore provides an integration with [OpenID connect](https://openid.net/connect/) services. Having this support properly configured, you can make MapStore users able to login with the given OpenID service.
 "

File: docs/developer-guide/integrations/users/openId.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+
+## Customizing logo an text in Login Form
+
+For details about the configuration for a specific service, please refer to the specific section below. For details about `authenticationProviders` optional values (e.g. to customize icon and/or text to show), refer to the documentation of the LoginPlugin.
+
+By default `authenticationProviders` is `{"type": "basic", "provider": "geostore"}`, that represents the standard login on mapstore with username and password. With the default configuration, when the user try to login MapStore will show the classical login form.
+
+Tou can add other providers to the list (e.g. `openid`) and they will be added as options to the login window.
```
Comment (tdipisa) [inline docs/developer-guide/integrations/users/openId.md:12]: It is possible to add other ...

File: docs/developer-guide/integrations/users/openId.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+You can remove the `geostore` entry from `authenticationProviders` list to remove the login form from the possible login systems.
+
+!!! note
+    If only one openID entry is present in `authenticationProviders` (and no `geostore` entry), clicking on login in the login menu will not show any intermediate window and you will be redirected directly to the only openID provider configured. If more than one entry is present in `authenticationProviders` list, the user will have to choose which one is ok.
```
Comment (tdipisa) [inline docs/developer-guide/integrations/users/openId.md:16]: If only one OpenID entry is present in `authenticationProviders` (and no `geostore` entry available), clicking on login in the login menu will not show any intermediate window and you will be redirected directly to the OpenID provider configured. If more than one entry is present in `authenticationProviders` list, the user will have to choose one of them to be authenticated.

**Review by tdipisa — CHANGES_REQUESTED** [review]
@offtherailz @taba90 
This is my review for the documentation part. Let me know if you need a sync up for this.

File: docs/developer-guide/integrations/users/openId.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+- Using the standard OpenID protocol
+- Supporting SSO (not yet implemented)
+- Integrating with users and roles, as well as for ldap. (not yet implemented)
+
+In this section you can see how to configure keycloak as a standard OpenID provider
+
+#### Configure keycloak Client
+
+Create a new Client on keycloak. In this guide we will name it `mapstore-server` (because if you need to configure SSO, we may need another key to call `mapstore-client`)
```
Comment (MV88) [inline docs/developer-guide/integrations/users/openId.md:110]: "we may need another key to call `mapstore-client"

can you elaborate a bit more here?

**Review by MV88 — CHANGES_REQUESTED** [review]
apart from tobias comment I have just this one

**Comment by ale-cristofori** [comment]
this issue is connected #8192, this PR does not cover all ACs for that issue. Outstanding is the SSO part which is being currently developed by @offtherailz

File: docs/developer-guide/integrations/users/openId.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+!!! note
+    If only one openID entry is present in `authenticationProviders` (and no `geostore` entry), clicking on login in the login menu will not show any intermediate window and you will be redirected directly to the only openID provider configured. If more than one entry is present in `authenticationProviders` list, the user will have to choose which one is ok.
+
+## Supported OpenID services
+
+MapStore allows to integrate with the following OpenID providers.
+
+- Google
+- Keycloak (to be implemented)
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply docs/developer-guide/integrations/users/openId.md:23]: forgot to clear

File: docs/developer-guide/integrations/users/openId.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    ]
+}
+```
+
+### Keycloak
+
+[Keycloak](https://www.keycloak.org/) is an open source identity and access management application widely used. MapStore has the ability to integrate with keycloak:
+
+- Using the standard OpenID protocol
+- Supporting SSO (not yet implemented)
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply docs/developer-guide/integrations/users/openId.md:103]: This PR contains the OpenID support, not the SSO support

File: docs/developer-guide/integrations/users/openId.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+- Integrating with users and roles, as well as for ldap. (not yet implemented)
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply docs/developer-guide/integrations/users/openId.md:104]: I'd like to link the proper documentation section when this has been implemented

File: docs/developer-guide/integrations/users/openId.md (hunk 5 above)
Comment (offtherailz) [reply docs/developer-guide/integrations/users/openId.md:110]: It will have a dedicated section in doc when inplemented.

**Comment by offtherailz** [comment]
@tdipisa I merged after fixing documentation accordingly to your review. The rest of the documentation will be completed with direct integration

**Comment by offtherailz** [comment]
Build originally passed and this PR was merged successfully.

Anyway I'm trying to run another execution of tests to check an issue on master and confirm that this build (successfully when merged 3 days ago) today fails, and so deduce that something changed outside mapstore code in the meanwhile. 
Just reporting this in case in the future we notice an execution error on last commit.

**Pull request merged**